src/main/java/de/tum/cit/aet/artemis/core/repository/CourseRepository.java

```diff
@@ -109,6 +109,15 @@ SELECT COUNT(c) > 0
     @EntityGraph(type = LOAD, attributePaths = { "exercises", "exercises.categories", "exercises.teamAssignmentConfig" })
     Course findWithEagerExercisesById(long courseId);
 
+    @Query("""
+            SELECT c
+            FROM Course c
+                LEFT JOIN FETCH c.exercises exercises
+                WHERE c.id = :courseId
+                    AND (exercises.releaseDate <=:now  OR exercises.releaseDate IS NULL)
+            """)
+    Optional<Course> findWithEagerReleasedExercisesById(@Param("courseId") long courseId, @Param("now") ZonedDateTime now);
```

Comment:
The tests fail because this query doesn't work if the course has **ONLY** exercises that are **unreleased**. Then, no row is returned. However, it should still return a row with only the course columns populated in this case.

In the test, that happens because the ProgrammingExerciseFactory sets the exercises for tests by default with an release date in _now() + 1 day_.